trufflesuite/truffle
# Attempt to fix types on contract-schema

## PR description

This PR demonstrates a conceptual approach to closing #5746 by broadening the string-indexed type to allow for undefined, which is needed for the TypeScript type declaration file generator, as that generator marks all the specified "properties" above as optional.   Another strategy could be to not mark those specified properties as optional, but at first glance that seems more likely to break things.
 
I don't expect this to actually work because `undefined` isn't a primitive type in the vocabulary of the [instance data model](https://json-schema.org/draft/2020-12/json-schema-core.html#name-instance-data-model). 

## Testing instructions
Run steps involved in building/preparing for release and inspect contract-schema/spec/index.d.ts to see if it includes the `|undefined` possibility that the Issue notes fixes the issue when manually added in.

## Documentation


## Breaking changes and new features

This should just broaden a type and not break existing code that uses narrower types, unless that code is somehow relying on the optional string-index value to be defined for any given string.
